platform/README.md

```diff
@@ -1,20 +1,223 @@
+```
+TALI_OPTIONAL_WORDS := [
+    "ed", "editor", "ramdrive", "block",
+    "environment?", "assembler", "disassembler", "wordlist"
+]
+```
+The currently available groups are:
+
+- `ed` is a string editor. (~1.5K)
+- `editor` is a block editor. The EDITOR-WORDLIST will also be removed. (~0.25K)
+- `ramdrive` is for testing block words without a block device. (~0.3K)
+- `block` is the optional BLOCK words. (~1.4K)
+- `environment?` is the ENVIRONMENT? word.  While this is a core word
+   for ANS-2012, it uses a lot of strings and therefore takes up a lot of memory. (~0.2K)
+- `assembler` is a 65c02 assembler.
+   The ASSEMBLER-WORDLIST will also be removed if the assembler is removed. (~3.2K)
+- `disassembler` is the disassembler word DISASM. (~0.6K)
+   If both the assembler and dissasembler are removed, the tables
+   (used for both assembling and disassembling) will be removed
+   for additional memory savings. (extra ~1.6K)
+- `wordlist` is for the optional SEARCH-ORDER words (eg. wordlists)
+   Note: Without "wordlist", you will not be able to use any words from
+   the EDITOR or ASSEMBLER wordlists (they should probably be disabled
+   by also removing "editor" and "assembler"), and all new words will
+   be compiled into the FORTH wordlist. (~0.9K)
+
+You can control what character(s) are printed by the word
+CR in order to move the cursor to the next line.  The default is "lf"
+for a line feed character (#10).  "cr" will use a carriage return (#13).
+Having both will use a carriage return followed by a line feed.  This
+only affects output.  Either CR or LF can be used to terminate lines
+on the input.
+```
+TALI_OPTION_CR_EOL := [ "lf" ]
+; TALI_OPTION_CR_EOL := [ "cr" ]
+; TALI_OPTION_CR_EOL := [ "cr", "lf" ]
+```
+By default Tali provides editable input history buffers via ctrl-n/ctrl-p.
+These can be disabled by setting `TALI_OPTION_HISTORY` to 0, saving about ~0.2K of Tali ROM and 1K RAM.
+```
+TALI_OPTION_HISTORY := 1
+; TALI_OPTION_HISTORY := 0      ; disable history
+```
+Finally, setting `TALI_OPTION_TERSE` to 1 strips or shortens various strings to reduce the memory
+footprint, saving about ~0.5K.
+```
+TALI_OPTION_TERSE := 0
+; TALI_OPTION_TERSE := 1        ; shorten/remove strings
+```
+
+### Kernel I/O routines
+
+Tali needs platform specific I/O routines to get input from the user
+and display output back to them.  It expects the platform configuration
+to provide the following kernel routines:
+
+- `kernel_init` Initialize the low-level hardware.  This is normally also
+  the reset vector.  It should set up your hardware, print the startup
+  message and `jmp forth`.
+- `kernel_getc` Get a single character from the keyboard and return
+  it in the accumulator.  It should block until a character is ready,
+  and should preserve the X and Y registers.
+- `kernel_putc` Send the character in A to the output device (e.g. screen).
+  It should preserve the X and Y registers. [TODO: does it need to preserve A as well?]
```

Comment:
question: does kernel_putc need to preserve the accumulator?

Reply:
No, kernel_putc does not need to preserve A.  We had an issue with this where `SPACES` only loaded A once, so that routine was modified to reload A for every (indirect) call to kernel_putc.  I checked all of the other locations where I/O is done (there actually aren't that many) and I believe it's fine everywhere else.